<a href="https://colab.research.google.com/github/AAFREEN-SAYYED/ML-Practical-Solution/blob/gh-pages/ML_5_Practical.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Multiple Regression**

Multiple regression is like linear regression, but with more than one independent value, meaning that we try to predict a value based on two or more variables.



In [ ]:
#The Pandas module allows us to read csv files and return a DataFrame object.
import pandas
import matplotlib.pyplot as plt

df = pandas.read_csv("cars.csv")

'''
Then make a list of the independent values and call this variable X.
Put the dependent values in a variable called y.
'''
X = df[['Weight', 'Volume']]
y = df['CO2']

'''
From the sklearn module we will use the LinearRegression() method to create a linear regression object.

This object has a method called fit() that takes the independent and dependent values as parameters
and fills the regression object with data that describes the relationship:
'''
from sklearn import linear_model

regr = linear_model.LinearRegression()
regr.fit(X, y)

#Now we have a regression object that are ready to predict CO2 values based on a car's weight and volume:

predictedCO2 = regr.predict([[2300, 1300]])
print(predictedCO2)
#We have predicted that a car with 1.3 liter engine, and a weight of 2300 kg, will release approximately 107 grams of CO2 for every kilometer it drives.


[107.2087328]


**Coefficient**

The coefficient is a factor that describes the relationship with an unknown variable.


Example: if x is a variable, then 2x is x two times. x is the unknown variable, and the number 2 is the coefficient.


In this case, we can ask for the coefficient value of weight against CO2, and for volume against CO2. The answer(s) we get tells us what would happen if we increase, or decrease, one of the independent values.



In [ ]:
print(regr.coef_)

[0.00755095 0.00780526]


**Result Explained**

The result array represents the coefficient values of weight and volume.

Weight: 0.00755095
Volume: 0.00780526

These values tell us that if the weight increase by 1kg, the CO2 emission increases by 0.00755095g.

And if the engine size (Volume) increases by 1 cm3, the CO2 emission increases by 0.00780526 g.

I think that is a fair guess, but let test it!

We have already predicted that if a car with a 1300cm3 engine weighs 2300kg, the CO2 emission will be approximately 107g.

What if we increase the weight with 1000kg?



In [ ]:
#change the weight from 2300 to 3300:
predictedCO2 = regr.predict([[3300, 1300]])

print(predictedCO2)

[114.75968007]


We have predicted that a car with 1.3 liter engine, and a weight of 3300 kg, will release approximately 115 grams of CO2 for every kilometer it drives.

Which shows that the coefficient of 0.00755095 is correct:

107.2087328 + (1000 * 0.00755095) = 114.75968

